### PR TITLE
Document methods for the BoneAttachment and IP classes

### PR DESCRIPTION
**Pull request by ghost**
Unclear if these need to be documented due to lack of replacement member variables. Documenting just in case.

### CODE REVIEW

File: doc/base/classes.xml [hunk 1]
```diff
@@ -22141,7 +22141,7 @@
 			<argument index="0" name="id" type="int">
 			</argument>
 			<description>
-				Returns a resolved item address if one is available. Returns an empty string on error or if resolution hasn't happened yet (see [method get_resolve_item_status]).
+			git	Returns a resolved item address if one is available. Returns an empty string on error or if resolution hasn't happened yet (see [method get_resolve_item_status]).
```
Comment (djrm) [inline doc/base/classes.xml]: ? :)

File: doc/base/classes.xml [hunk 2]
```diff
@@ -22107,7 +22107,7 @@
 		IP Protocol support functions.
 	</brief_description>
 	<description>
-		IP contains support functions for the IPv4 protocol. TCP/IP support is in different classes (see [StreamPeerTCP] and [TCP_Server]). IP provides hostname resolution support, both blocking and threaded.
+        IP contains support functions for the IPv4 protocol. TCP/IP support is in different classes (see [StreamPeerTCP] and [TCP_Server]). IP provides hostname resolution support, both blocking and threaded.
```
Comment (djrm) [inline doc/base/classes.xml]: this was actually ok

File: doc/base/classes.xml (hunk 1 above)
Comment (deliciousfudge) [reply doc/base/classes.xml]: Haha whoops, just seen that. Will fix that up too.

**Comment by djrm** [comment]
also it would be great if you could squash yout commits into one, try

```
git rebase -i HEAD~3
```

and put `s` instead of `pick` in the last two commits.

**Comment by deliciousfudge** [comment]
Thanks, will do that now once I've double-checked the indents

File: doc/base/classes.xml [hunk 3]
```diff
@@ -22122,13 +22125,14 @@
 			<argument index="0" name="id" type="int">
 			</argument>
 			<description>
-				Erase a queue ID, removing it from the queue if needed. This should be used after a queue is completed to free it and enable more queries to happen.
+                		Erases a queue ID, and removes it from the queue if needed. This should be used to free a queue after it has completed to enable more queries to happen.
```
Comment (NathanLovato) [inline doc/base/classes.xml]: Indent issue: convert all indents to tabs

File: doc/base/classes.xml [hunk 4]
```diff
@@ -22137,7 +22141,7 @@
 			<argument index="0" name="id" type="int">
 			</argument>
 			<description>
-				Return a resolved item address, or an empty string if an error happened or resolution didn't happen yet (see [method get_resolve_item_status]).
+				Returns a resolved item address if one is available. Returns an empty string on error or if resolution hasn't happened yet (see [method get_resolve_item_status]).
```
Comment (NathanLovato) [inline doc/base/classes.xml]: What's a resolved item address? As a developer, you'll read the name of the method, which often tells you some of what it does. The description should help you understand the details.
NB: if you explained item addresses it in the class description, it's okay - tell the user to "read the class description above for more information", or you may re-explain it in short here

File: doc/base/classes.xml [hunk 5]
```diff
@@ -22146,7 +22150,7 @@
 			<argument index="0" name="id" type="int">
 			</argument>
 			<description>
-				Return the status of hostname queued for resolving, given its queue ID. Returned status can be any of the RESOLVER_STATUS_* enumeration.
+				Returns the status of the hostname queued for resolving, given its queue ID. Returned status can be any of the RESOLVER_STATUS_* enumerators.
```
Comment (NathanLovato) [inline doc/base/classes.xml]: `Returned status can be any of the RESOLVER_STATUS_* enumerators.` -> `Returns one of the RESOLVER_STATUS_* constants.`
`enumerators -> enum`

**Review by NathanLovato — COMMENTED** [review]
The docs lack clarity. It's not due to your work in particular: the original descriptions are obscure, they assume you understand a lot of concepts. I've done some web development, and I still don't know what the methods do from reading their name and the descriptions.

File: doc/base/classes.xml [hunk 6]
```diff
@@ -9261,6 +9261,7 @@
 			<return type="String">
 			</return>
 			<description>
+                		Returns the name of the bone node attached to.
```
Comment (Noshyaar) [inline doc/base/classes.xml]: Indentation issue here. Should be 4 tabs.

File: doc/base/classes.xml [hunk 7]
```diff
@@ -22105,7 +22107,7 @@
 		IP Protocol support functions.
 	</brief_description>
 	<description>
-		IP contains some support functions for the IPv4 protocol. TCP/IP support is in different classes (see [StreamPeerTCP] and [TCP_Server]). IP provides hostname resolution support, both blocking and threaded.
+        	IP contains support functions for the IPv4 protocol. TCP/IP support is in different classes (see [StreamPeerTCP] and [TCP_Server]). IP provides hostname resolution support, both blocking and threaded.
```
Comment (Noshyaar) [inline doc/base/classes.xml]: Indentation issue here. Should be 2 tabs.

**Comment by Noshyaar** [comment]
There are some indentation issues. Please use tabs to indent and squash commits together.

**Comment by deliciousfudge** [comment]
Have pushed the requested changes through. Hopefully that sorts things.

**Pull request merged**